api/src/test/java/org/apache/iceberg/util/TestCharSequenceSet.java

```diff
@@ -79,4 +81,35 @@ public void testRemoveAll() {
 
     Assertions.assertThat(set).isEmpty();
   }
+
+  @Test
+  public void testEqualsAndHashCode() {
+    CharSequenceSet set1 = CharSequenceSet.empty();
+    CharSequenceSet set2 = CharSequenceSet.empty();
+
+    Assertions.assertThat(set1).isEqualTo(set2);
+    Assertions.assertThat(set1.hashCode()).isEqualTo(set2.hashCode());
+
+    set1.add("v1");
+    set1.add("v2");
+    set1.add("v3");
+
+    set2.add(new StringBuilder("v1"));
+    set2.add(new StringBuffer("v2"));
+    set2.add("v3");
+
+    Set<CharSequence> set3 = Collections.unmodifiableSet(set2);
```

Comment:
I am primarily interested in wrapping `CharSequenceSet` into `UnmodifiableSet` in `CharSequenceMap`.